[E2E][ST-Engine] Add test data consistency in 3 node cluster and fix bug

EricJoy2048 opened the pull request:
closes https://github.com/apache/incubator-seatunnel/issues/3049

EricJoy2048 committed:
add test data consistency in 3 node cluster

add data consistency test in 3 node cluster

EricJoy2048 committed:
Merge remote-tracking branch 'apache/dev' into 4_add_test_data_count

EricJoy2048 committed:
Fix FakeSource generate data have row separator and column separator

Hisoka-X commented on seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/FileUtils.java:
```diff
@@ -80,4 +84,77 @@ public static void createNewFile(String filePath) {
+     */
+    public static Long getFileLineNumber(@NonNull String filePath) {
+        try (LineNumberReader lineNumberReader = new LineNumberReader(new FileReader(filePath))) {
+            lineNumberReader.skip(Long.MAX_VALUE);
```
Why we should skip some char?

EricJoy2048 replied:
> 

I will use replace LineNumberReader, because if the file content chars more than Long.MAX_VALUE, it will error.

```
try {
        return Files.lines(Paths.get(filePath)).count();
    } catch (IOException e) {
        return -1;
    }
```

Hisoka-X commented on seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/FileUtils.java:
```diff
@@ -80,4 +84,77 @@ public static void createNewFile(String filePath) {
+        try (LineNumberReader lineNumberReader = new LineNumberReader(new FileReader(filePath))) {
+            lineNumberReader.skip(Long.MAX_VALUE);
+            long lineNumber = lineNumberReader.getLineNumber();
+            return lineNumber + 1;
```
Why should +1?

EricJoy2048 committed:
use Files.lines get the line numbers of file

use Files.lines get the line numbers of file

use Files.lines get the line numbers of file

EricJoy2048 committed:
fix conflict

EricJoy2048 committed:
fix filepath replaceAll in windows error

fix filepath replaceAll in windows error

EricJoy2048 committed:
fix ci error

CalvinKirs commented on seatunnel-common/src/main/java/org/apache/seatunnel/common/utils/FileUtils.java:
```diff
@@ -79,4 +82,76 @@ public static void createNewFile(String filePath) {
+            if (file.isDirectory()) {
+                deleteFiles(file);
+            }
+            file.delete();
```
Here it may fail, we'd better add the corresponding judgment, if it fails, print the corresponding log (if the failure result is not so important)

CalvinKirs reviewed (approved):
LGTM
